tools/jit/gen_unboxing_wrappers.py

```diff
@@ -135,6 +136,7 @@ def jit_type_of(arg):
     'int64_t': '{}.toInt()',
     'int64_t?': '{}.toOptional<int64_t>()',
     'std::string': '{}.toStringRef()',
+    'std::string?': '{}.toOptional<std::string>()',
```

Comment:
Following the mapping for `std::string`, I think it should be`'std::string?': '{}.toOptionalStringRef()'`

Reply:
After some experimenting, I think that would require the C++ signature of native functions be changed from `c10::optional<std::string>` to `c10::optional<std::reference_wrapper<const std::string>>` which I think is really ugly.

A nicer way to avoid the copy would be to change from `std::string` to `c10::string_view` everywhere, in which case we don't need the reference_wrapper.

Reply:
But I figure the copy is okay since `std::string` functions will take a copy currently anyway. So I think a `string_view` conversion could be left for a separate PR.